lti_consumer/tests/unit/plugin/test_views_lti_ags.py

```diff
@@ -274,3 +301,556 @@ def test_create_lineitem_invalid_resource_link_id(self):
+        primary_user_results_endpoint = reverse(
+            'lti_consumer:lti-ags-view-results',
+            kwargs={
+                "lti_config_id": self.lti_config.id,
+                "pk": self.line_item.id,
+                "user_id": self.primary_user_id
+            }
+        )
+        secondary_user_results_endpoint = reverse(
+            'lti_consumer:lti-ags-view-results',
+            kwargs={
+                "lti_config_id": self.lti_config.id,
+                "pk": self.line_item.id,
+                "user_id": self.secondary_user_id
+            }
+        )
+        self.assertEqual(
+            [dict(d) for d in response.data],
+            [
+                {
+                    "id": "http://testserver" + primary_user_results_endpoint,
+                    "scoreOf": "http://testserver" + self.lineitem_endpoint,
+                    "userId": self.primary_user_id,
+                    "resultScore": 83.0,
+                    "resultMaximum": 100.0,
+                    "comment": "This is exceptional work."
+                },
+                {
+                    "id": "http://testserver" + secondary_user_results_endpoint,
+                    "scoreOf": "http://testserver" + self.lineitem_endpoint,
+                    "userId": self.secondary_user_id,
+                    "resultScore": 25.0,
+                    "resultMaximum": 100.0,
+                    "comment": "This is not great work."
+                }
+            ]
+        )
+
+    def test_retrieve_results_for_user_id(self):
+        """
+        Test the LTI AGS LineItem Score Creation.
+        """
+        self._set_lti_token('https://purl.imsglobal.org/spec/lti-ags/scope/result.readonly')
+
+        results_user_endpoint = reverse(
+            'lti_consumer:lti-ags-view-results',
+            kwargs={
+                "lti_config_id": self.lti_config.id,
+                "pk": self.line_item.id,
+                "user_id": self.secondary_user_id
+            }
+        )
+
+        # Request results with userId
+        response = self.client.get(results_user_endpoint, data={"userId": self.secondary_user_id})
+
+        self.assertEqual(response.status_code, 200)
+
+        # There should be 1 result for that user
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]['userId'], self.secondary_user_id)
+
+    def test_retrieve_results_with_limit(self):
+        """
+        Test the LTI AGS LineItem Score Creation.
+        """
+        self._set_lti_token('https://purl.imsglobal.org/spec/lti-ags/scope/result.readonly')
+
+        # Request results with limit
+        response = self.client.get(self.results_endpoint, data={"limit": 1})
+
+        self.assertEqual(response.status_code, 200)
+
+        # There should be 1 results, and it should be the one with the latest timestamp
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]['userId'], self.primary_user_id)
```

Comment:
This doesn't check everything the comment mentions: "the one with the latest timestamp".

Reply:
@nedbat 

Unfortunately, the results service does not return any timestamp information. As such, I can't make any assertions of that kind. I will add comments clarifying this though.